src/features/filter.ts

```diff
@@ -1,12 +1,28 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
-const initialState = {
+type Filter = {
+  query: string;
+  status: Status;
+};
+
+const initialState: Filter = {
```

Comment:
The initial state for 'status' is set to 'all', but 'Status' is likely an enum or a type that might not include 'all'. Ensure that 'Status' includes 'all' or adjust the initial state to match a valid 'Status' value.